docs/proposals/02-backupinfra.md

```diff
@@ -0,0 +1,232 @@
+    name: packet
+spec:
+  provider: aws
+  region: eu-west-1
+  secretRef: # Required for root
+    name: backup-operator-aws
+    namespace: backup-garden
+status:
+  lastOperation: ...
+  observedGeneration: ...
+  seed: ...
+```
+
+```yaml
+apiVersion: garden.cloud/v1alpha1
+kind: BackupEntry
+metadata:
+  name: shoot--dev--example--3ef42 # Naming convention explained before
+  namespace: garden-dev
+  ownerReferences:
+  - apiVersion: garden.sapcloud.io/v1beta1
+    blockOwnerDeletion: false
+    controller: true
+    kind: Shoot
+    name: example
+    uid: 19a9538b-5058-11e9-b5a6-5e696cab3bc8
+spec:
+  shootUID: 19a9538b-5058-11e9-b5a6-5e696cab3bc8 # Just for reference to find back associated shoot.
+  # Following section comes from cloudProfile or seed yaml based on granularity decision.
+  provider: aws
+  region: eu-west-1
+  secretRef: # Required for root
+    name: backup-operator-aws
+    namespace: backup-garden
+status:
+  lastOperation: ...
+  observedGeneration: ...
+  seed: ...
+```
+
+### CRD on Seed cluster
+Considering the extension proposal, we want individual component to be handled by controller inside seed cluster. We will have Backup related resource in registered seed cluster as well.
+
+```yaml
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: BackupContainer
+metadata:
+  name: packet-random[:5]
+spec:
+  type: aws
+  region: eu-west-1
+  secretRef:
+    name: backup-operator-aws
+    namespace: backup-garden
+status:
+  observedGeneration: ...
+  state: ...
+  lastError: ..
+  lastOperation: ...
+```
+
+There are two points for introducing BackupEntry resource.
+1. Cloud provider specific code goes completely in seed cluster.
+2. Network issue is also handled by moving deletion part to backup-extension-controller in seed cluster.
+
+```yaml
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: BackupEntry
+metadata:
+  name: shoot--dev--example--3ef42 # Naming convention explained later
+  namespace: backup-garden
+spec:
+  type: aws
+  region: eu-west-1
+  secretRef: # Required for root
+    name: backup-operator-aws
+    namespace: backup-garden
+status:
+  observedGeneration: ...
+  state: ...
+  lastError: ..
+  lastOperation: ...
+```
+
+### Workflow
+
+- Gardener administrator will configure the cloudProfile with backup infra credentials and provider config as follows.
+
+```yaml
+# CloudProfile.yaml:
+Spec:
+    backup:
+        provider: aws
+        region: eu-qest-1
+        secretRef:
+            name: backup-operator-aws
+            namespace: garden
+```
+Here CloudProfileController will interpret this spec as follows:
+- If `spec.backup.region` is not nil,
+  - then respect it, i.e. use the provider and unique region field mentioned there for backupContainer.
+  - Here Preferably, `spec.backup.region` field will be unique, since for cross provider, it doesn’t make much sense. Since region name will be different for different providers.
+- Otherwise, spec.backup.region is nil then,
```

Comment:
This is confusing because you are talking about "spec.backup.provider = spec.(type of provider)" in the next line ;-) I got what you mean, but please adapt it so that it's easier understandable.